### ISSUE TITLE
twisted.test.test_process.ProcessTestCase.testManyProcesses fails with little information

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5652]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5652|
|Type|enhancement|
|Created|2012-04-22 15:31:20Z|
|Branch|https://github.com/twisted/twisted/tree/process-test-fail-with-info-5652|

`testManyProcesses` has been a problem on Windows for years now.  Part of the challenge of fixing it is that it's completely opaque as to why it fails.  The exception that fails the test is a bare `RuntimeError`.  By reading the test, the only thing it's possible to infer is that "abcd" was not received on standard output.

As a first tiny step towards fixing this test, it would be nice to see what _is_ received on standard output when this test fails.

<details><summary>Searchable metadata</summary>

```
trac-id__5652 5652
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_process_test_fail_with_info_5652 branches/process-test-fail-with-info-5652
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__windows_tests windows tests
time__1335108680000000 1335108680000000
changetime__1337085690000000 1337085690000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [34254]) Branching to 'process-test-fail-with-info-5652'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [34256]) Add some information to the RuntimeErrors rudely raised during testManyProcesses

refs [#5652](https://github.com/twisted/twisted/issues/5652)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Very minor change, hopefully it'll make some Windows failed builds more interesting.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @itamarst|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

If you're going to be adding err to RuntimeError, you can drop the print. Other than that, looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [34409]) Drop the print

refs [#5652](https://github.com/twisted/twisted/issues/5652)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [34410]) Merge process-test-fail-with-info-5652

Author: exarkun
Reviewer: itamar
Fixes: [#5652](https://github.com/twisted/twisted/issues/5652)

Add more information to the `testManyProcesses` failure, in hopes of
aiding future debugging.
